### PR TITLE
Add support for TIBCO's Avro serializer/deserializer

### PR DESCRIPTION
**Pull request by ebowden-tibco**
This PR works as-is for me for deserializing & displaying messages that were serialized by the TIBCO Avro serialization library.  Since the TIBCO client library isn't currently available via a public Maven repository, it must be manually downloaded (as part of the "TIBCO® Messaging - Schema Repository for Apache Kafka - Community Edition" package at [https://www.tibco.com/products/tibco-messaging/downloads](https://www.tibco.com/products/tibco-messaging/downloads)) and then AKHQ must be started via setting the classpath with the TIBCO jar and its dependencies specified in addition to AKHQ's jar, rather than just using the -jar option.  Meaning - with this PR, instead of starting AKHQ like:

```
java -Dmicronaut.config.files=/path/to/config.yml -jar /path/to/akhq-0.1-all.jar
```

You'll need to start it by calling:

```
java -Dmicronaut.config.files=/path/to/config.yml -cp /opt/tibco/akd/repo/1.2/lib/tibftl-kafka-avro-1.2.0-thin.jar:/path/to/akhq-0.1-all.jar:/opt/tibco/akd/repo/1.2/lib/deps/retrofit-2.9.0.jar:/opt/tibco/akd/repo/1.2/lib/deps/converter-gson-2.9.0.jar:/opt/tibco/akd/repo/1.2/lib/deps/okhttp-3.14.9.jar:/opt/tibco/akd/repo/1.2/lib/deps/okio-jvm-2.9.0.jar:/opt/tibco/akd/repo/1.2/lib/deps/kotlin-stdlib-1.4.10.jar org.akhq.App
```

Additionally, in your AKHQ config file's 'schema-registry' section, you'll need to add a new 'type' option set to 'tibco' if you're using a TIBCO schema repository with the TIBCO serializer/deserializer.  E.g.:

```
akhq:
  connections:
    dev:
      properties:
        bootstrap.servers: "localhost:9092"
        security.protocol: plaintext
      schema-registry:
        url: "http://localhost:8585/schema/v1"
        type: "tibco"
        properties: {}
      connect:
        - name: connect-1
```

(Also note the URL; it should be set to the URL of the TIBCO FTL Server used by the TIBCO schema repository with "/schema/v1" appended at the end as shown above.)

This PR contains just a minimal number of changes to allow full compatibility with the TIBCO schema repository and Avro serializer client library.  It may be worth discussing more substantial changes to AKHQ to allow for the flexibility to specify a SerDe class and options to use on a per-topic-regex basis (similar to what's currently used for the JSON deserializer) in the future, which would allow AKHQ to support any serialization format out there that has an existing Kafka SerDe for it without needing to add explicit support for the format to AKHQ each time.

### CODE REVIEW

**Comment by ebowden-tibco** [comment]
Hi there - just checking in to see if anyone's had a chance to take a look at this PR.  Let me know what you think - what you'd like changed if anything, etc.  Thanks!

File: src/main/java/org/akhq/configs/Connection.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public Connection(@Parameter String name) {
         String url;
         String basicAuthUsername;
         String basicAuthPassword;
+        String type;
```
Comment (tchiotludo) [inline src/main/java/org/akhq/configs/Connection.java:34]: Change this one to an Enum please and put a default one to Confluent

File: src/main/java/org/akhq/repositories/SchemaRegistryRepository.java [hunk 2]
```diff
@@ -221,20 +225,38 @@ public void updateConfig(String clusterId, String subject, Schema.Config config)
...
+                    Map<String, String> config = new HashMap<>();
+                    config.put("schema.registry.url", this.kafkaModule.getConnection(clusterId).getSchemaRegistry().getUrl());
+                    if (this.kafkaModule.getConnection(clusterId).getSchemaRegistry().getBasicAuthUsername() != null) {
+                        config.put("ftl.username", this.kafkaModule.getConnection(clusterId).getSchemaRegistry().getBasicAuthUsername());
+                        config.put("ftl.password", this.kafkaModule.getConnection(clusterId).getSchemaRegistry().getBasicAuthPassword());
+                    }
+                    config.putAll(this.kafkaModule.getConnection(clusterId).getProperties());
+                    deserializer.configure(config, false);
+                } else {
+                    deserializer = (Deserializer) Class.forName("io.confluent.kafka.serializers.KafkaAvroDeserializer").getDeclaredConstructor(SchemaRegistryClient.class).newInstance(this.kafkaModule.getRegistryClient(clusterId));
```
Comment (tchiotludo) [inline src/main/java/org/akhq/repositories/SchemaRegistryRepository.java:244]: Why use of reflection here ? 
The class is in class path

**Review by tchiotludo — CHANGES_REQUESTED** [review]

**Comment by tchiotludo** [comment]
sorry @ebowden-tibco , completely forgot. 
Comment with some minor change. 

Also you need to provide : 
- some documentation on README.md (maybe you need also to have the instruction for helm charts, how to mount jar) .
- populate the [application example](https://github.com/tchiotludo/akhq/blob/dev/application.example.yml) with the new configuration. 
- add a new configuration_env variable for the classpath (`-cp`) for the docker version [here](https://github.com/tchiotludo/akhq/blob/dev/docker/app/akhq)

And it will be fine :+1: 
Thanks

**Comment by ebowden-tibco** [comment]
Thanks for the feedback!  I'll update the PR soon with the requested changes.

**Comment by tchiotludo** [comment]
Perfect !! 
Thanks @ebowden-tibco

**Comment by AntAreS24** [comment]
@tchiotludo, out of curiosity, could I modify the build.gradle to add a new Maven repo (company Nexus) https://github.com/tchiotludo/akhq/blob/aada01e36407a3cc1799263c0996fb100905829a/build.gradle#L33
with the TIBCO Jar in it, and add the reference to the TIBCO jar https://github.com/tchiotludo/akhq/blob/aada01e36407a3cc1799263c0996fb100905829a/build.gradle#L107?

I will still need to make the changes for the schema repository type in the config file.

**Comment by tchiotludo** [comment]
I don't think so, the license will not allow to embed a proprietary jar on this opensource project. 
(I think I don't the license of TIBCO)

**Comment by AntAreS24** [comment]
Of course not here, but if I clone it locally and make the changes.

**Comment by xpr3sso** [comment]
Hi @ebowden-tibco ,
do I need some config adjustments to run your changes without having a registry? :)
Can't access the topic list for my local test cluster with these changes..

**Comment by ebowden-tibco** [comment]
@xpr3sso - You shouldn't need any config changes.  Could you give an example config snippet that fails for you? Do you just not have a "schema-registry" section in your config at all?  Does everything work immediately before commit aada01e36407a3cc1799263c0996fb100905829a ? ie, if you checkout at commit 477c7046790e1b8371ca73da37df82c375e360bb, do things work again? I'm not sure how these changes would've affected AKHQ use without a registry, but I'm new to the AKHQ code base, so anything's possible.

**Comment by xpr3sso** [comment]
@ebowden-tibco just checked https://github.com/tchiotludo/akhq/commit/477c7046790e1b8371ca73da37df82c375e360bb and this is working fine for me :)
The injection of Connection.SchemaRegistry (https://github.com/tchiotludo/akhq/pull/558/files#diff-3901f50bbc040388cf793601f793a114ab4bb6d3fe0dd15f06c33dec41e3d12bR61) seems to fail, as I don't have any specified

> Internal Server Error: Failed to inject value for field [schemaRegistry] of class: org.akhq.repositories.RecordRepository Path Taken: TopicController.recordRepository --> RecordRepository.schemaRegistry
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for field [schemaRegistry] of class: org.akhq.repositories.RecordRepository
Path Taken: TopicController.recordRepository --> RecordRepository.schemaRegistry
	at io.micronaut.context.AbstractBeanDefinition.getBeanForField(AbstractBeanDefinition.java:1462)
	at io.micronaut.context.AbstractBeanDefinition.injectBeanField(AbstractBeanDefinition.java:731)
	at org.akhq.repositories.$RecordRepositoryDefinition.injectBean(Unknown Source)
	at org.akhq.repositories.$RecordRepositoryDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForField(AbstractBeanDefinition.java:1441)
	at io.micronaut.context.AbstractBeanDefinition.injectBeanField(AbstractBeanDefinition.java:731)
	at org.akhq.controllers.$TopicControllerDefinition.injectBean(Unknown Source)
	at org.akhq.controllers.$TopicControllerDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.access$100(DefaultBeanContext.java:78)
	at io.micronaut.context.DefaultBeanContext$4.getTarget(DefaultBeanContext.java:431)
	at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:474)
	at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:312)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$10(RoutingInBoundHandler.java:1363)
	at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)
	at io.micronaut.scheduling.instrument.InvocationInstrumenterWrappedRunnable.run(InvocationInstrumenterWrappedRunnable.java:47)
	at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:79)
	at io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:148)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [org.akhq.configs.Connection$SchemaRegistry] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2290)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForField(AbstractBeanDefinition.java:1441)
	... 35 more
`

**Comment by ebowden-tibco** [comment]
Doh!  Yes, I can reproduce it here.  I did all my testing with a schema-registry section in the config file.  Tested with and without the new schema registry "type" option present, but didn't think to test with/without the entire schema-registry config block itself.  I'll try to get this fixed today.

**Pull request merged**